app/controllers/names_controller.rb

```diff
@@ -121,7 +121,10 @@ def names_with_descriptions
 
   # Display list of names that a given user is author on.
   def names_by_user
-    user = params[:id] ? find_or_goto_index(User, params[:by_user].to_s) : @user
+    user = find_obj_or_goto_index(
+      model: User, obj_id: params[:by_user].to_s,
+      index_path: names_path
+    )
```

Comment:
Changes functionality when by_user is not an existing user.
See [Comment about LocationsController](https://github.com/MushroomObserver/mushroom-observer/pull/1324/files#r1095161355).